backend/src/test/java/org/uresti/pozarreal/service/impl/HousesServiceImplTests.java

```diff
@@ -0,0 +1,223 @@
+    }
+
+    @Test
+    public void givenAnHouseId_thenToggleChipStatusRequest() {
+        // Given:
+        HousesRepository housesRepository = Mockito.mock(HousesRepository.class);
+        StreetRepository streetRepository = Mockito.mock(StreetRepository.class);
+        RepresentativeRepository representativeRepository = Mockito.mock(RepresentativeRepository.class);
+        SessionHelper mockSessionHelper = Mockito.mock(SessionHelper.class);
+        HousesServiceImpl housesService = new HousesServiceImpl(housesRepository, streetRepository, representativeRepository, mockSessionHelper);
+
+        org.uresti.pozarreal.model.House house = new org.uresti.pozarreal.model.House();
+        house.setId("1");
+        house.setNotes("hello");
+        house.setStreet("Street 1");
+        house.setNumber("123456");
+        house.setChipsEnabled(true);
+
+        Street street = new Street();
+
+        street.setId("1");
+        street.setName("name 1");
+
+        Mockito.when(housesRepository.save(house)).thenReturn(house);
+        Mockito.when(streetRepository.findById(house.getStreet())).thenReturn(java.util.Optional.of(street));
+        Mockito.when(housesRepository.findById("1")).thenReturn(java.util.Optional.of(house));
+
+        // When:
+        housesService.toggleChipStatusRequest("1", false);
+        HouseInfo houseInfo = housesService.getHouseInfo("1");
+
+        // Then:
+        Assertions.assertThat(houseInfo.getNotes()).isEqualTo("hello");
+        Assertions.assertThat(houseInfo.getStreet()).isEqualTo("Street 1");
+        Assertions.assertThat(houseInfo.getNumber()).isEqualTo("123456");
+        Assertions.assertThat(houseInfo.getId()).isEqualTo("1");
+        Assertions.assertThat(houseInfo.isChipsEnabled()).isFalse();
+    }
+
+    @Test
+    public void givenAnCorrectHouseId_thenReturnHouseInfo() {
+        //Given:
+        HousesRepository housesRepository = Mockito.mock(HousesRepository.class);
+        StreetRepository streetRepository = Mockito.mock(StreetRepository.class);
+        RepresentativeRepository representativeRepository = Mockito.mock(RepresentativeRepository.class);
+        SessionHelper mockSessionHelper = Mockito.mock(SessionHelper.class);
+        HousesServiceImpl housesService = new HousesServiceImpl(housesRepository, streetRepository, representativeRepository, mockSessionHelper);
+
+        List<org.uresti.pozarreal.model.House> houses = new LinkedList<>();
+
+        org.uresti.pozarreal.model.House house = new org.uresti.pozarreal.model.House();
+        house.setId("1");
+        house.setNotes("hello");
+        house.setStreet("123");
+        house.setNumber("123456");
+        house.setChipsEnabled(true);
+        houses.add(house);
+
+        Street street = new Street();
+
+        street.setId("1");
+        street.setName("name 1");
+
+        // When:
+        Mockito.when(housesRepository.findAllByStreetOrderByNumber("123")).thenReturn(houses);
```

Comment:
The "When" section is used to locate the method that you're testing. 

All these sentences "Mockito.when" are part of test initialization and belongs to section "given".

In this case the only line that belongs to "when" section is 
```
HouseInfo houseInfo = housesService.getHouseInfo("1");
```

Reply:
done